### DISCUSSION TITLE
When Muon wins and when it doesn't: a spectral-tail threshold

### DISCUSSION THREAD

**Discussion by idevender** [KellerJordan/modded-nanogpt#352] [Show and tell]
Muon is exact on truly low-rank targets and degrades fastest as the target grows a spectral tail, ceding to GD near 4% tail energy in this diagnostic (underdetermined matrix sensing, 10 seeds, float64, all methods interpolating). Recent work disagrees about Muon's bias - some report a strong spectral simplicity bias, others report it fitting spurious features in deep-linear models. Sweeping the tail shows both on one axis when you control for training loss.

<img width="1298" height="694" alt="fig4_phase" src="https://github.com/user-attachments/assets/7ed6b7dc-aec0-4d84-be74-9bb96a2add3e" />

The predictor I found is whether the update is invariant to the rotation $(U,V) \to (UQ,VQ)$ that the loss can't see in a factored model $W = UV^T$. Adam's per-coordinate second moment isn't. GD, shared-scalar Adam, Muon, and Shampoo are. Within the invariant group, the ordering follows schedule aggressiveness: Muon most aggressive, then Shampoo, then GD. Aggressive wins accurate low-rank recovery; moderate wins robustness to a tail that shouldn't be fit.

This isn't a speed claim - modded-nanogpt has shown Muon wins on speed and that's orthogonal. It's a CPU diagnostic for screening optimizer bias before spending H100 hours, especially if you're testing a new variant.

**Paper:** [https://arxiv.org/abs/2608.05136](https://arxiv.org/abs/2608.05136)

**Code, logs, seeds:** [https://github.com/idevender/loss-basis-adam](https://github.com/idevender/loss-basis-adam)
